pennylane/transforms/compile.py

```diff
@@ -140,14 +140,15 @@ def qfunc(x, y, z):
     .. code-block:: python3
 
         compiled_qfunc = qml.compile(
+            qfunc,
```

Comment:
Should we actually demonstrate applying the transform to the qnode instead? That's the way I would recommend applying transforms, as it will probably be more performant and it will integrate better with different components.

Reply:
We can do that although it would make this block slightly inconsistent with the one above and more verbose.

Compare:
```python
@partial(
    compile,
    pipeline=[
        partial(qml.transforms.commute_controlled, direction="left"),
        partial(qml.transforms.merge_rotations, atol=1e-6),
        qml.transforms.cancel_inverses
    ],
    basis_set=["CNOT", "RX", "RY", "RZ"],
    num_passes=2,
)
@qml.qnode(device=dev)
def compiled_qnode(x, y, z):
    return qfunc(x, y, z)
```
to
```python
compiled_qfunc = qml.compile(
    qfunc,
    pipeline=[
        partial(qml.transforms.commute_controlled, direction="left"),
        partial(qml.transforms.merge_rotations, atol=1e-6),
        qml.transforms.cancel_inverses
    ],
    basis_set=["CNOT", "RX", "RY", "RZ"],
    num_passes=2
)

compiled_qnode = qml.QNode(compiled_qfunc, dev)
```
Which would you prefer, @albi3ro? Should we also change the example above to use the declarative style?

Reply:
Potentially:
```
uncompiled_qnode = qml.QNode(qfunc, dev)

compiled_qnode = qml.compile(
    uncompiled_qnode,
    pipeline=[
        partial(qml.transforms.commute_controlled, direction="left"),
        partial(qml.transforms.merge_rotations, atol=1e-6),
        qml.transforms.cancel_inverses
    ],
    basis_set=["CNOT", "RX", "RY", "RZ"],
    num_passes=2
)
```

Reply:
Ah, sorry, I understand what you mean now. Will fix!

Reply:
Should be good now, @albi3ro!